authors/apps/articles/views.py

```diff
@@ -119,3 +124,78 @@ class ArticleRetrieveBySlugAPIView(generics.RetrieveAPIView):
     renderer_classes = (ArticleRenderer,)
     permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
     lookup_field = 'slug'
+
+
+class LikeAPIView(generics.GenericAPIView):
+    """
+    post: Create an  like or dislike.
+    """
+    serializer_class = LikeSerializer
+    permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
+    renderer_classes = (LikeStatusRenderer,)
+    lookup_field = 'pk'
+
+    def post(self, request, *args, **kwargs):
+        if not Article.objects.filter(pk=kwargs['pk']).exists():
+            raise NotFound(detail="Article Not found", code=404)
+        article_id = kwargs['pk']
+        like_status = request.data.get('like_status')
+        user, author_username = get_id_from_token(request)
+
+        if LikeArticle.objects.filter(user=user).filter(article=article_id).exists():
+            raise NotAcceptable(
+                detail="You have already liked/disliked this article",
+                code=406)
+        article_data = Article.objects.get(pk=article_id)
+        new_like_status = {
+            "article": article_id,
+            "article_title": article_data.title,
+            "like_status": like_status,
+            "user": user
+        }
+        serializer = self.serializer_class(data=new_like_status)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    def get(self, request, *args, **kwargs):
+        if not Article.objects.filter(pk=kwargs['pk']).exists():
+            raise NotFound(detail="Article Not found", code=404)
+        if LikeArticle.objects.filter(article=kwargs['pk']).exists():
+            like_status = LikeArticle.objects.filter(article=kwargs['pk'])
+            like_statuses = LikeSerializer(like_status, many=True)
+            return Response(like_statuses.data, status=status.HTTP_200_OK)
+        return Response(
+            {"error": "This article has not been liked/disliked yet"},
+            status=status.HTTP_400_BAD_REQUEST)
+
+
+class UpdateLikeStatusAPIView(generics.RetrieveUpdateAPIView):
+    serializer_class = LikeSerializer
+    permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
+    renderer_classes = (LikeStatusRenderer,)
+    lookup_field = 'pk'
+
+    def put(self, request, *args, **kwargs):
+        article_id = kwargs['pk']
+        if not Article.objects.filter(pk=article_id).exists():
+            raise NotFound(detail="Article Not found", code=404)
```

Comment:
No need for this `code=404` Read about exceptions.

Reply:
Done